languagetool-language-modules/ga/src/main/java/org/languagetool/tagging/ga/Utils.java

```diff
@@ -0,0 +1,390 @@
+    if(in.charAt(0) != first && in.charAt(0) != upperFirst) {
+      return null;
+    }
+    // properly eclipsed
+    if(in.charAt(0) == first && (in.charAt(1) == second || in.charAt(1) == upperSecond)) {
+      return in.substring(1);
+    }
+
+    char ch1 = in.charAt(1);
+    if(in.length() > 3 && in.charAt(1) == '-') {
+      from++;
+      ch1 = in.charAt(2);
+    }
+    if(ch1 == second || ch1 == upperSecond) {
+      return Character.toString(retSecond)+ in.substring(from);
+    } else {
+      return null;
+    }
+  }
+
+  public static boolean isUpperVowel(char c) {
+    switch(c) {
+      case 'A':
+      case 'E':
+      case 'I':
+      case 'O':
+      case 'U':
+      case '\u00c1':
+      case '\u00c9':
+      case '\u00cd':
+      case '\u00d3':
+      case '\u00da':
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  public static boolean isLowerVowel(char c) {
+    switch(c) {
+      case 'a':
+      case 'e':
+      case 'i':
+      case 'o':
+      case 'u':
+      case '\u00e1':
+      case '\u00e9':
+      case '\u00ed':
+      case '\u00f3':
+      case '\u00fa':
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  public static boolean isVowel(char c) {
+    return isLowerVowel(c) || isUpperVowel(c);
+  }
+
+  public static boolean isUpperLenitable(char c) {
+    switch(c) {
+      case 'B':
+      case 'C':
+      case 'D':
+      case 'F':
+      case 'G':
+      case 'M':
+      case 'P':
+      case 'S':
+      case 'T':
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  public static boolean isLowerLenitable(char c) {
+    switch(c) {
+      case 'b':
+      case 'c':
+      case 'd':
+      case 'f':
+      case 'g':
+      case 'm':
+      case 'p':
+      case 's':
+      case 't':
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  public static boolean isSLenitable(char c) {
+    switch(c) {
+      case 'l':
+      case 'n':
+      case 'r':
+      case 'a':
+      case 'e':
+      case 'i':
+      case 'o':
+      case 'u':
+      case '\u00e1':
+      case '\u00e9':
+      case '\u00ed':
+      case '\u00f3':
+      case '\u00fa':
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  public static String lenite(String in) {
+    if(in.length() < 2) {
+      return in;
+    }
+    String outh = (Character.isUpperCase(in.charAt(0)) && Character.isUpperCase(1)) ? "H" : "h";
+    if(isLowerLenitable(in.charAt(0)) || isUpperLenitable(in.charAt(0))) {
+      if(in.charAt(0) == 'S' || in.charAt(0) == 's') {
+        if(isSLenitable(Character.toLowerCase(in.charAt(1)))) {
+          return Character.toString(in.charAt(0)) + outh + in.substring(1);
+        } else {
+          return in;
+        }
+      } else {
+        return Character.toString(in.charAt(0)) + outh + in.substring(1);
+      }
+    } else {
+      return in;
+    }
+  }
+
+  public static String eclipse(String in) {
```

Comment:
Short javadoc would be nice to explain what "eclipse" mean in GA.

Reply:
tl;dr - 'to apply eclipsis': there's a rather lengthy Wikipedia article about Irish initial mutations: https://en.wikipedia.org/wiki/Irish_initial_mutations
(oh, and I've added the javadoc)